Packages/ClientRuntime/Sources/Config/SDKRuntimeConfiguration.swift

```diff
@@ -17,6 +17,7 @@ public protocol SDKRuntimeConfiguration {
     var logger: LogAgent {get}
     var clientLogMode: ClientLogMode {get}
     var retryer: SDKRetryer {get}
+    var endpoint: String? {get set}
```

Comment:
**Question**: Why is a simple string sufficient for configuration? Won't you need an endpoint _resolver_ at the client level which returns an `Endpoint` struct (not a simple string)?

Reply:
This is to allow customers configure their endpoint by themselves. This is not particularly useful for AWS SDK, but required for generic SDKs, because `EndpointResolver` is not part of the base configuration anymore.

When the endpoint resolution happens, this string will be considered and prioritized.

Reply:
This seems far less flexible than allowing a custom endpoint resolver. It only allows for a single URL regardless of operation, which may not be sufficient. It also doesn't allow for custom headers.

Reply:
Fair call, I have asked this question with endpoints 2.0 team. I agree, this is limiting and ideally we should allow overriding `Endpoint` type than just a `String` `endpoint`.

However, this would have ripple effect to other places if we want resolve endpoint at a common place which is ideal but we can short circuit if endpoint exists to reduce the scope of changes.